keras_nlp/samplers/greedy_sampler.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Greedy Sampler."""
+
+import tensorflow as tf
+
+from keras_nlp.samplers.sampler import Sampler
+from keras_nlp.samplers.sampler import base_sampler_keyword_args
+from keras_nlp.samplers.sampler import call_keyword_docstring
+from keras_nlp.samplers.sampler import sample_keyword_docstring
+
+
+class GreedySampler(Sampler):
+    """Greedy Sampler class.
+
+    This sampler is implemented on greedy search, i.e., always picking up the
+    token of the largest probability as the next token.
+
+    Args:
+        {{base_sampler_keyword_args}}
```

Comment:
This feels like a step to far towards conciseness instead of readability. I would write these out for each of our samplers, to keep our code much more readable. E.g. take a look at [Conv2D](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/layers/convolutional/conv2d.py#L85) and [Conv3D](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/layers/convolutional/conv3d.py#L66).

Reply:
I can see this either way. @chenmoneygithub pointed out that this was the pattern he used for keras optimizers with @fchollet input. On the other hand, we don't expect too many samplers so this won't be the same maintainability problem as it is for tasks where we might literally have hundreds with the same copypasta.

Reply:
Followup thought: don't we expect most people to read our docstrings on keras.io? What proportion is going to be reading the raw code?